### ISSUE TITLE
MacOS 经验分享

### ISSUE THREAD

**Issue by Yufeng98** [OSCPU/ysyx#5]
### Verilator

Fail to compile from source code in GitHub Repositary. Use `brew install verilator` to install 4.200 version, slightly behind the latest 4.204 version, but also works well.

#### Avoid Security Check

Add `--no-quarantine` to avoid security protection on MacOS. Use `brew install gtkwave --no-quarantine` to install GTKWave.

#### Strange Error

> /usr/local/bin/gtkwave: line 122: test: too many arguments
> /usr/local/bin/gtkwave: line 149: test: ==: unary operator expected
find: /share/locale: No such file or directory
> /usr/local/bin/gtkwave: line 215: /usr/local/bin/../../../Contents/Resources/bin/: No such file or directory
> /usr/local/bin/gtkwave: line 215: exec: /usr/local/bin/../../../Contents/Resources/bin/: cannot execute: No such file or directory

[GTKWave on macOS](https://ughe.github.io/2018/11/06/gtkwave-osx) gives a good solution. 

**TL;DR**

Use Perl’s package manager to install Switch and run binary with full path.

```bash
cpan install Switch
perl -V:'installsitelib'
/Applications/gtkwave.app/Contents/Resources/bin/gtkwave
```

### Adjust Build Script

In `build.sh`, some commands only works in Linux.

1. There is no `-f` option with `readlink` in MacOS. Use `SHELL_PATH=$(pwd)` to show current directory.

2. There is no `hostnamectl` command in MacOX. Use `hostname` instead.

3. Substitute `gtkwave` with full path `/Applications/gtkwave.app/Contents/Resources/bin/gtkwave`

**Comment by ghost** [comment]
关于macos15的分享：
- nemu找不到.so文件：关注macOS上的动态库格式。
- homebrew的gtkwave弃用：去该代码仓库的相关issues的找找。
